### PR TITLE
feat: add `--Xbenchmark-incremental` compiler flag

### PR DESCRIPTION
**Pull request by mlutze**
Part of #3166

### CODE REVIEW

**Comment by magnus-madsen** [comment]
Maybe the first compilation run should be free?

**Comment by magnus-madsen** [comment]
Are the numbers averages or what? What happens if you use your quick-n-dirty timer?

**Comment by magnus-madsen** [comment]
Also try to rebase onto master.

**Comment by mlutze** [comment]
Fixed it! Typer isn't exactly zero, but there's at least a significant difference:

```
matt@matt-laptop:~/Downloads$ java -jar flix.jar --Xbenchmark-phases; java -jar flix.jar --Xbenchmark-incremental
====================== Flix Compiler Phases ======================              

Best runtime per phase (low to high):
  Deriver                          1.3 ms (00.0%)
  Reader                           1.3 ms (00.0%)
  Terminator                       4.9 ms (00.1%)
  Safety                           7.1 ms (00.1%)
  Instances                        7.2 ms (00.2%)
  Tailrec                          8.0 ms (00.2%)
  Stratifier                       9.6 ms (00.2%)
  VarNumbering                    14.8 ms (00.3%)
  Eraser                          16.0 ms (00.3%)
  Simplifier                      17.5 ms (00.4%)
  Lowering                        18.1 ms (00.4%)
  Optimizer                       19.1 ms (00.4%)
  PatternExhaustiveness           21.7 ms (00.5%)
  LambdaLift                      21.8 ms (00.5%)
  Kinder                          30.1 ms (00.6%)
  ClosureConv                     53.1 ms (01.1%)
  TreeShaker                      55.1 ms (01.2%)
  Namer                           84.8 ms (01.8%)
  Redundancy                      87.1 ms (01.8%)
  Weeder                          96.9 ms (02.0%)
  Finalize                       106.2 ms (02.2%)
  Parser                         247.6 ms (05.2%)
  Resolver                       248.5 ms (05.2%)
  Monomorph                      556.5 ms (11.7%)
  JvmBackend                     1236.3 ms (26.0%)
  Typer                          1792.5 ms (37.6%)

Finished 10 iterations on 52,027 lines of code in 51 seconds.
====================== Flix Compiler Phases ======================              

Best runtime per phase (low to high):
  Reader                           1.3 ms (00.0%)
  Deriver                          1.3 ms (00.0%)
  Terminator                       5.7 ms (00.1%)
  Instances                        6.9 ms (00.2%)
  Safety                           7.1 ms (00.2%)
  Tailrec                          8.0 ms (00.2%)
  Stratifier                       8.7 ms (00.2%)
  VarNumbering                    14.1 ms (00.4%)
  Lowering                        14.7 ms (00.4%)
  Simplifier                      17.3 ms (00.4%)
  Eraser                          17.8 ms (00.4%)
  PatternExhaustiveness           18.7 ms (00.5%)
  Optimizer                       19.3 ms (00.5%)
  LambdaLift                      22.1 ms (00.5%)
  Kinder                          26.4 ms (00.7%)
  Weeder                          44.0 ms (01.1%)
  ClosureConv                     52.8 ms (01.3%)
  TreeShaker                      57.3 ms (01.4%)
  Namer                           66.9 ms (01.7%)
  Redundancy                      79.6 ms (02.0%)
  Parser                         107.9 ms (02.7%)
  Finalize                       111.4 ms (02.8%)
  Resolver                       211.7 ms (05.2%)
  Monomorph                      541.5 ms (13.4%)
  JvmBackend                     1233.1 ms (30.5%)
  Typer                          1340.9 ms (33.2%)

Finished 10 iterations on 52,027 lines of code in 36 seconds.
```

**Comment by magnus-madsen** [comment]
What was the issue?

**Comment by mlutze** [comment]
> What was the issue?

Forgot to re-add all the inputs. The `ChangeSet` remained `Everything`.

File: main/src/ca/uwaterloo/flix/api/Flix.scala [hunk 1]
```diff
@@ -36,6 +36,12 @@ object Flix {
     * The reserved Flix delimiter.
     */
   val Delimiter: String = "%"
+
+
+  /**
+    * A case class to track the compile time spent in a phase and its sub-phases.
+    */
+  case class PhaseTime(phase: String, time: Long, subphases: List[(String, Long)])
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/Flix.scala:44]: This is now public, so lets move it into the api directory proper. I don't like to refer to Flix.PhaseTime.

**Comment by magnus-madsen** [comment]
Thanks!- now for my part.

**Pull request merged**